runtime/v2/model/src/main/java/com/walmartlabs/concord/runtime/v2/model/ExclusiveMode.java

```diff
@@ -65,6 +60,19 @@ enum Mode {
     }
 
     static ExclusiveMode of(String group, Mode mode) {
-        return ImmutableExclusiveMode.of(group, mode);
+        return ImmutableDefaultExclusiveMode.of(group, mode);
+    }
+
+    @Value.Immutable
+    @Value.Style(jdkOnly = true)
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @JsonSerialize(as = ImmutableDefaultExclusiveMode.class)
+    @JsonDeserialize(as = ImmutableDefaultExclusiveMode.class)
+    interface DefaultExclusiveMode extends ExclusiveMode {
```

Comment:
Not sure why do we need a separate interface here?

Reply:
One `@Immutable` class should not extends another `@Immutable` class.
```java
@Value.Immutable
public interface MyInterface {
    boolean isTrue();
}

@Value.Immutable
public interface AnotherInterface extends MyInterface {
    boolean isFalse();
}
```

```
AnotherInterface.java: (immutables:subtype) Should not inherit com.example.MyInterface which is
a value type itself. Avoid extending from another abstract value type. Better to share common
abstract class or interface which are not carrying @Immutable annotation. If still extending from
immutable abstract type be ready to face some incoherences in generated types.
```

Reply:
> Does it change the serialization of TriggerMixIn?

Is the concern in this question regarding somewhere we deserialize it from a value stored in the db? i.e. a value stored pre-upgrade may fail after this change is in?

Reply:
I misunderstood what was happening here, my bad.

Reply:
I'm still not quite sure why a new class appeared here...

Wouldn't it be enough to just change:

GithubTriggerMixIn:
```
            @JsonProperty("exclusive")
            ExclusiveMode exclusive();
```
to
```
            @JsonProperty("exclusive")
            GithubTriggerExclusiveMode exclusive();
```

?

Reply:
something like this:
```
@@ -111,7 +112,7 @@ public interface TriggerMixIn extends Trigger {
             GithubTriggerConditions conditions();
 
             @JsonProperty("exclusive")
-            ExclusiveMode exclusive();
+            GithubTriggerExclusiveMode exclusive();
 
             interface GithubTriggerConditions {
                 @JsonProperty(value = "type", required = true)
@@ -174,6 +175,9 @@ public interface TriggerMixIn extends Trigger {
 
             @JsonProperty("conditions")
             Map<String, Object> conditions();
+
+            @JsonProperty("exclusive")
+            ExclusiveMode exclusive();
         }
     }
 
@@ -187,8 +191,5 @@ public interface TriggerMixIn extends Trigger {
 
         @JsonProperty("arguments")
         Map<String, Object> arguments();
-
-        @JsonProperty("exclusive")
-        ExclusiveMode exclusive();
     }
 }
```

Reply:
Not sure if that would be reflected in the resulting schema?
Currently, it's 
```json
"triggers" : {
      "type" : "array",
      "items" : {
        "oneOf" : [ {
          "$ref" : "#/definitions/GenericTrigger"
        }, {
          "$ref" : "#/definitions/CronTrigger"
        }, {
          "$ref" : "#/definitions/ManualTrigger"
        }, {
          "$ref" : "#/definitions/GithubTrigger"
        }, {
          "$ref" : "#/definitions/OneOpsTrigger"
        } ]
      }
}
```
With GithubTrigger containing

```json
"GithubTriggerParams" : {
      "type" : "object",
      "additionalProperties" : false,
      "properties" : {
        "version" : {
          "type" : "integer",
          "enum" : [ 2 ]
        },
        "exclusive" : {
          "$ref" : "#/definitions/GithubTriggerExclusiveMode"
        },
        ...
```

and other trigger types using the base ExclusiveMode:

```json
"ManualTriggerParams" : {
      "type" : "object",
      "additionalProperties" : false,
      "properties" : {
        "name" : {
          "type" : "string"
        },
        "arguments" : {
          "type" : "object"
        },
        "activeProfiles" : {
          "type" : "array",
          "items" : {
            "type" : "string"
          }
        },
        "entryPoint" : {
          "type" : "string"
        },
        "exclusive" : {
          "$ref" : "#/definitions/ExclusiveMode"
        }
      },
      "required" : [ "entryPoint" ]
    },
```

Which seems fine to me?

Reply:
> Not sure if that would be reflected in the resulting schema?
It should be, why not?

from master:
```
...
"GithubTriggerParams" : {
...
        "exclusive" : {
          "$ref" : "#/definitions/ExclusiveMode"
        },
....
```
withi just ExclusiveMode -> GithubTriggerExclusiveMode change:
```
...
"GithubTriggerParams" : {
...
        "exclusive" : {
          "$ref" : "#/definitions/GithubTriggerExclusiveMode"
        },
....

    "GithubTriggerExclusiveMode" : {
      "type" : "object",
      "additionalProperties" : false,
      "properties" : {
        "group" : {
          "type" : "string"
        },
        "groupBy" : {
          "type" : "string"
        },
        "mode" : {
          "type" : "string",
          "enum" : [ "cancel", "cancelOld", "wait" ]
        }
      }
    }
```